src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCacheableRequest.java

```diff
@@ -491,101 +297,126 @@ private boolean storeRequest(
+        case CACHE_ENTRY_NOT_MODIFIED_SIGNAL:
+            onResponseSignalCacheEntryNotModified(signal);
+            break;
+        case CACHE_ENTRY_UPDATED_SIGNAL:
+            onResponseSignalCacheEntryUpdated(signal);
+            break;
+        case CACHE_ENTRY_ABORTED_SIGNAL:
+            onResponseSignalCacheEntryAborted(signal);
+            break;
+        default:
+            break;
+        }
+    }
 
-    private void purge()
+    private void onResponseSignalCacheEntryUpdated(
+        SignalFW signal)
     {
-        if (requestSlot.value != NO_SLOT)
+        cacheEntry = factory.defaultCache.get(requestGroup.getRequestHash());
+        if (payloadWritten == -1)
         {
-            factory.requestBufferPool.release(requestSlot.value);
-            requestSlot.value = NO_SLOT;
+            resetRequestTimeoutIfNecessary();
+            sendHttpResponseHeaders(cacheEntry);
+        }
+        else
+        {
+            final long traceId = signal.traceId();
+            factory.budgetManager.resumeAssigningBudget(budgetId, 0, traceId);
+            sendEndIfNecessary(traceId);
         }
-        isRequestPurged = true;
     }
 
-    private void incAttempts()
+    private void onResponseSignalRequestExpired(
+        SignalFW signal)
     {
-        attempts++;
+        factory.defaultCache.send304(ifNoneMatch,
+                                     prefer,
+                                     accept,
+                                     acceptRouteId,
+                                     acceptReplyId);
+        requestExpired = true;
     }
 
-    private Consumer<ArrayFW.Builder<HttpHeaderFW.Builder, HttpHeaderFW>> mutateRequestHeaders(
-        ArrayFW<HttpHeaderFW> requestHeaders)
+    private void onResponseSignalCacheEntryAborted(
+        SignalFW signal)
     {
-        return (ArrayFW.Builder<HttpHeaderFW.Builder, HttpHeaderFW> builder) ->
+        if (this.payloadWritten >= 0)
         {
-            requestHeaders.forEach(h ->
+            factory.writer.doAbort(accept,
+                                   acceptRouteId,
+                                   acceptReplyId,
+                                   signal.traceId());
+            requestGroup.dequeue(ifNoneMatch, acceptReplyId);
+            cacheEntry.setSubscribers(-1);
+        }
+        else
+        {
+            send503RetryAfter();
+        }
+        cleanupRequestIfNecessary();
+    }
+
+    private void onResponseSignalRequestGroupLeaderUpdated(
+        SignalFW signal)
+    {
+        final ArrayFW<HttpHeaderFW> requestHeaders = getRequestHeaders();
+        Consumer<ArrayFW.Builder<HttpHeaderFW.Builder, HttpHeaderFW>> mutator =
+            builder -> requestHeaders.forEach(h -> builder.item(item -> item.name(h.name()).value(h.value())));
+
+        connect = factory.writer.newHttpStream(requestGroup::newRequest, connectRouteId, connectInitialId,
+            factory.supplyTraceId.getAsLong(), mutator,
+            (t, b, i, l) ->
```

Comment:
Please change this lambda to a method reference, say `onConnectMessage` that detects `ResetFW.TYPE_ID` and hands off to `onResponseReset` method with implementation to propagate `RESET` and `cleanupRequestIfNecessary()`.

Reply:
Done.